src/App.tsx

```diff
@@ -1,24 +1,133 @@
+// eslint-disable-next-line max-len
+import NotificationError from './components/NotificationError/NotificationError';
 
-const USER_ID = 0;
+const USER_ID = 6982;
 
 export const App: React.FC = () => {
+  const [task, setTask] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState(ErrorTypes.none);
+  const [tempTodo, setTemptodo] = useState<Todo | null>(null);
+  const [isDataUpdated, setIsdataUpdated] = useState(false);
+  const [activeIds, setActiveIds] = useState([0]);
+
+  const handleTodoChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setTask(event.target.value);
+  };
+
+  const resetTempTodo = useCallback(() => {
+    setTemptodo(null);
+    setActiveIds([0]);
+  }, []);
+
+  const loadTodos = async () => {
+    setIsdataUpdated(true);
+
+    try {
+      const todosFromServer = await getTodos(Number(USER_ID));
+
+      setTodos(todosFromServer);
+    } catch {
+      setError(ErrorTypes.get);
+    } finally {
+      setIsdataUpdated(false);
+      resetTempTodo();
+    }
+  };
+
+  const handleTodoSubmit = async (event: ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!task.trim().length) {
+      setError(ErrorTypes.emptyTitle);
+      setTask('');
+
+      return;
+    }
+
+    setTemptodo({
+      id: 0,
+      userId: USER_ID,
+      title: task,
+      completed: false,
+    });
+    setTask('');
+
+    try {
+      await addTodo({
+        userId: USER_ID,
+        title: task,
+        completed: false,
+      });
+      loadTodos();
+    } catch {
+      setError(ErrorTypes.add);
+      resetTempTodo();
+    }
+  };
+
+  const handleRemoveTodo = async (id: number) => {
+    setActiveIds((activeId) => [...activeId, id]);
+
+    try {
+      await removeTodo(id);
+      loadTodos();
+    } catch {
+      setError(ErrorTypes.delete);
+      resetTempTodo();
+    }
+  };
+
+  const handleClearCompleted = () => {
+    todos.forEach((todo) => {
+      if (todo.completed) {
+        handleRemoveTodo(todo.id);
+      }
+    });
+  };
+
+  const resetError = () => setError(ErrorTypes.none);
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title is-1">todos</h1>
+
+      <div className="todoapp__content">
+        <NewTodo
+          task={task}
+          isDataUpdated={isDataUpdated}
+          handleTodoChange={handleTodoChange}
+          handleTodoSubmit={handleTodoSubmit}
+        />
+
+        {(todos.length > 0 || tempTodo) && (
```

Comment:
Here you can use just `todos.length` instead of `todos.length > 0` [see the doc](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)